contracts/router/SynapseRouterV2.sol

```diff
@@ -0,0 +1,361 @@
+            if (!does) delete unflattened[i];
+            else count += pool.tokens.length;
+        }
+
+        // flatten into supported tokens and filter out duplicates
+        supportedTokens = unflattened.flatten(count).unique();
+
+        // add native weth if in supported list
+        if (supportedTokens.contains(swapQuoter.weth())) {
+            return supportedTokens.append(UniversalTokenLib.ETH_ADDRESS);
+        }
+    }
+
+    /// @inheritdoc IRouterV2
+    function getDestinationAmountOut(DestRequest[] memory requests, address tokenOut)
+        external
+        view
+        returns (SwapQuery[] memory destQueries)
+    {
+        destQueries = new SwapQuery[](requests.length);
+        for (uint256 i = 0; i < requests.length; ++i) {
+            DestRequest memory request = requests[i];
+            (address token, uint256 actionMask, address bridgeModule) = _getTokenAndActionMaskFromSymbol(
+                request.symbol
+            );
+            if (token == address(0)) continue;
+
+            // account for bridge fees in amountIn
+            bool isSwap = !(token == tokenOut ||
+                (tokenOut == UniversalTokenLib.ETH_ADDRESS && token == swapQuoter.weth()));
+            uint256 amountIn = _calculateBridgeAmountIn(bridgeModule, token, request.amountIn, isSwap);
+            if (amountIn == 0) continue;
+
+            // query the quoter
+            LimitedToken memory tokenIn = LimitedToken({actionMask: actionMask, token: token});
+            destQueries[i] = swapQuoter.getAmountOut(tokenIn, tokenOut, amountIn);
+        }
+    }
+
+    /// @inheritdoc IRouterV2
+    function getOriginAmountOut(
+        address tokenIn,
+        string[] memory tokenSymbols,
+        uint256 amountIn
+    ) external view returns (SwapQuery[] memory originQueries) {
+        originQueries = new SwapQuery[](tokenSymbols.length);
+        for (uint256 i = 0; i < tokenSymbols.length; ++i) {
+            (address tokenOut, , address bridgeModule) = _getTokenAndActionMaskFromSymbol(tokenSymbols[i]);
+            if (tokenOut == address(0)) continue;
+
+            // query the quoter
+            LimitedToken memory _tokenIn = LimitedToken({actionMask: ActionLib.allActions(), token: tokenIn});
+            SwapQuery memory query = swapQuoter.getAmountOut(_tokenIn, tokenOut, amountIn);
+
+            // check max amount can bridge
+            uint256 maxAmountOut = IBridgeModule(bridgeModule).getMaxBridgedAmount(tokenOut);
+            if (query.minAmountOut > maxAmountOut) continue;
+
+            // set in return array
+            originQueries[i] = query;
+        }
+    }
+
+    /// @notice Checks whether module ID has already been connected to router
+    function _hasModule(bytes32 moduleId) internal view returns (bool) {
+        return _bridgeModules.contains(uint256(moduleId));
+    }
+
+    /// @notice Searches all bridge modules to get the token address from the unique bridge symbol
```

Comment:
Let's add a note that it returns zero values for unknown symbols (rather than reverts)